docs/docs/service-discovery-load-balancing.md

```diff
@@ -6,15 +6,15 @@ title: Service Discovery & Load Balancing
 
 Once our app is up and running, we need a way to send outside traffic to it. If we're running multiple apps, they also need a way to find each other. We can use [Mesos-DNS](https://github.com/mesosphere/mesos-dns) for service discovery through the domain name system ([DNS](http://en.wikipedia.org/wiki/Domain_Name_System)). Mesos-DNS generates a hostname for each application running on Mesos and translates these names to the IP address and port on the machine currently running each application. Mesos-DNS is particularly useful if the apps are launched through multiple frameworks (not just Marathon). Check the Mesos-DNS [documentation and tutorials page](http://mesosphere.github.io/mesos-dns/) for further information. 
 
-An alternative way for service discovery is to run a TCP proxy on each host in the cluster, and transparently forward a static port on localhost to the hosts that are running the app. That way, clients simply connect to that port, and the implementation details of discovery are completely abstracted away. This approach is sufficient if all apps are launched through Marathon. 
+An alternative way for service discovery is to run a TCP proxy on each host in the cluster, and transparently forward connection to the static service port on localhost to the dynamically assigned host/port combinations of the individual app instances (= *tasks*). That way, clients simply connect to that port, and the implementation details of discovery are completely abstracted away. This approach is sufficient if all apps are launched through Marathon. 
 
-## Ports Assignment
+## Service Ports Assignment
 
-When an application is created in Marathon (either through the REST API or the front end) - you may assign it one or more ports. These ports may be any valid port number or 0, which Marathon will use to randomly assign a port number.
+When an application is created in Marathon (either through the REST API or the front end) - you may assign it one or more ports. These service ports may be any valid port number or 0 which Marathon will use to randomly assign a service port number. Marathon will ensure that all dynamically assigned service ports a unique for this Marathon instance to avoid overlaps. Marathon does not use these assigned service ports itself. It is the responsibility of a separate component to make use of this information and set up load balancing accordingly. A sample script that does this is provided with Marathon, the `haproxy-marathon-bridge`.
```

Comment:
Please default to active voice rather than passive voice.  E.g. "When you create an application" rather than "When an application is created"

> you may assign it one or more ports

This sounds like you're talking about the "ports" property, but really you're only talking about service ports + bridge networking.

Link to the "servicePort" section in the REST API, since that's what you're describing here.  And you should be clear that this only applies to docker + bridge networking.

Does "Marathon instance" mean a "task"?  If so, let's just call them "tasks".  Or does it mean "the marathon framework"?